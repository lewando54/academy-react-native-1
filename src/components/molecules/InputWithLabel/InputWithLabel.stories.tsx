/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react'
import { useState } from 'react'

import InputWithLabel from './InputWithLabel'
import { create } from 'react-test-renderer'

interface IInputWithLabelMockupProps {
    placeholder: string,
    value: string,
    secureTextEntry: boolean
    labelText: string
}

function StatefulInputWithLabel({placeholder, value, secureTextEntry, labelText}: IInputWithLabelMockupProps){
    const [textValue, setTextValue] = useState(value)

    return (
        <InputWithLabel labelText={labelText} value={textValue} placeholder={placeholder} secureTextEntry={secureTextEntry} onChange={(newValue) => setTextValue(newValue)}></InputWithLabel>
    )
}

const meta = {
    component: StatefulInputWithLabel,
    argTypes: {
        placeholder: { control: 'text' },
        value: { control: 'text' },
        secureTextEntry: { control: 'boolean' },
        labelText: { control: 'text' }
    }
} satisfies Meta<typeof StatefulInputWithLabel>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
    args: {
        placeholder: 'Type something',
        value: 'Initial value',
        secureTextEntry: false,
        labelText: 'Full name'
    }
}