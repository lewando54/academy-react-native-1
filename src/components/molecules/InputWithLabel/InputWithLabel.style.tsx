import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    div: {
        display: 'flex',
        width: '100%'
    },
    label: {
        fontSize: 16
    }
})