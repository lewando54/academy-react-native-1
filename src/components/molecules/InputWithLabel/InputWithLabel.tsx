import React, { forwardRef } from 'react'
import TextInputField, { IInputProps, ITextInputFieldRef } from '../../atoms/TextInputField/TextInputField'
import { View, Text } from 'react-native'
import inputWithLabelStyle from './InputWithLabel.style'

export interface IInputWithLabelProps extends IInputProps{
    labelText: string
}

function InputWithLabel({labelText, ...props}: IInputWithLabelProps, ref: React.Ref<ITextInputFieldRef>){
    return(
        <View style={inputWithLabelStyle.div}>
            <Text style={inputWithLabelStyle.label}>{labelText}</Text>
            <TextInputField {...props} ref={ref} />
        </View>
    )
}

export default forwardRef(InputWithLabel)