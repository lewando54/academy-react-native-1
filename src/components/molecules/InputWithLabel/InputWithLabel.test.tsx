import React from 'react'
import { render } from '@testing-library/react-native'
import * as renderer from 'react-test-renderer'
import InputWithLabel from './InputWithLabel'

describe('InputWithLabel', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<InputWithLabel labelText='Testing' testId='test' value='' onChange={() => {}}/>)
    expect(tree).toMatchSnapshot()
  })

  it('should render input with label with text', () => {
    const { getByText } = render(<InputWithLabel labelText='Testing' testId='test' value='' onChange={() => {}}/>)
    expect(getByText('Testing')).toBeTruthy()
  })
})
