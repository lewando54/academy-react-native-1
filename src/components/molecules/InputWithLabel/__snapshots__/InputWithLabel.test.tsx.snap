// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel should render correctly 1`] = `
<View
  style={
    {
      "display": "flex",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 16,
      }
    }
  >
    Testing
  </Text>
  <TextInput
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    secureTextEntry={false}
    style={
      [
        {
          "backgroundColor": "#fff",
          "borderColor": "#e5e7eb",
          "borderRadius": 6,
          "borderWidth": 2,
          "color": "black",
          "fontSize": 18,
          "lineHeight": 28,
          "padding": 4,
          "width": "100%",
        },
        false,
      ]
    }
    testID="test"
    value=""
  />
</View>
`;
